src/main/jbake/content/blog/2017/12/running-agent.adoc

```diff
@@ -0,0 +1,155 @@
+has had some changes. This document will discuss those changes and how to run the Hawkular Agent.
+
+First, the Hawkular Agent now only runs as a javaagent (no longer does it run inside WildFly as a subsystem extension).
+So to run the agent, simply attach it as a javaagent to whatever JVM you want to monitor (be it a WildFly or EAP app server or
+any JVM application that exposes metrics via JMX). For example, add this command line option to the command that starts the JVM:
+
+```
+-javaagent:hawkular-javaagent.jar=config=hawkular-javaagent-config.yaml,delay=10
+```
+
+The configuration file contains all the settings necessary to start the agent and connect to the Hawkular Server.
+Once connected, the agent will pull down additional configuration settings from the server.
+
+=== Running in WildFly
+
+If you are attaching the agent to a WildFly server, you must set some additional settings.
+
+If you are running WildFly in standalone mode, the standalone.conf you should have something like:
+
+```
+JBOSS_MODULES_SYSTEM_PKGS="org.jboss.byteman,org.jboss.logmanager"
+...
+JAVA_OPTS="$JAVA_OPTS -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
+...
+JAVA_OPTS="$JAVA_OPTS -javaagent:$JBOSS_HOME/bin/hawkular-javaagent.jar=config=$JBOSS_HOME/standalone/configuration/hawkular-javaagent-config.yaml,delay=10"
+
+```
+
+If you are running WildFly in domain mode, the domain.conf you should have something like:
+
+```
+HOST_CONTROLLER_JAVA_OPTS="$HOST_CONTROLLER_JAVA_OPTS -Dhawkular.agent.metadata.domain=true \
+   -Djboss.modules.system.pkgs=org.jboss.byteman,org.jboss.logmanager \
+   -Djava.util.logging.manager=org.jboss.logmanager.LogManager \
+   -Djboss.host.server-excluded-properties=jboss.modules.system.pkgs,java.util.logging.manager,hawkular.agent.metadata.domain \
+   -javaagent:$JBOSS_HOME/bin/hawkular-javaagent.jar=config=$JBOSS_HOME/domain/configuration/hawkular-javaagent-config-domain.yaml,delay=10"
+```
+
+==== Running in WildFly Domain Mode
+
+When running WildFly in domain mode, a Hawkular Agent runs in the host controller.
+There also needs to be a Hawkular Agent running in each of the slave servers spawned by the host controller. You ensure
+an agent is attached to each slave server by adding the proper -javaagent command line option to the
+server JVM command found in the host.xml. Note also that you also have to tell each slave server's agent what port it
+should bind the metrics exporter endpoint to. For example:
+
+```
+  <jvms>
+    <jvm name="default">
+      <jvm-options>
+        ...
+        <!-- HAWKULAR SETTINGS -->
+        <option value="-Djboss.modules.system.pkgs=org.jboss.byteman,org.jboss.logmanager"/>
+        <option value="-Djava.util.logging.manager=org.jboss.logmanager.LogManager"/>
+        <option value="-javaagent:${jboss.home.dir}/bin/hawkular-javaagent.jar=config=${jboss.domain.config.dir}/hawkular-javaagent-config-metrics-only.yaml,delay=10"/>
```

Comment:
Where can i find a sample of `hawkular-javaagent-config-metrics-only.yaml` ?

Reply:
I suppose is the regular yaml with only the 
```
metrics-exporter:
  config-file: WF10
```
 ?

Reply:
Found it here: https://github.com/josejulio/hawkular-agent/blob/953cda87dcd064ae6edd56da9e1e8fdc3493a19d/hawkular-javaagent-wildfly-feature-pack/src/main/resources/featurepack/content/standalone/configuration/hawkular-javaagent-config-metrics-only.yaml#L17-L16